### ISSUE TITLE
Return file content along with the diff via the Compare endpoint

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-server#14382]
Code Manager is currently written to base the code overview (the outline on the right-hand side) on the contents of the current version of a file, rather than the contents of the diff. This is a decision that was made long ago, and I imagine it was done to simplify rendering the overview, and to allow it to share code with the browse page. I believe there are also complications with the fact that a diff can contain multiple lines with the same line number.

What this means, however, is that every time we need to request a diff for a file (via the `/reviewers/addon/{addonId}/versions/{versionId}/compare_to/` endpoint), we also need to request the content for the same file (via the (via the `/reviewers/addon/{addonId}/versions/{versionId}/` endpoint). This is a significant waste of time and resources so we should fix it.

Two suggestions for fixing this are:

1. Have the `compare_to` API return the file content, in addition to the diff content. This is quite simple to do, as the serializer has access to it, but it would mean extra data, which could be significant for a very large file. This is what this issue is proposing to do.

2. Change the code in Code Manager to render the code overview based on the contents of the diff, rather than the contents of the file. This would be a more involved fix, and is something that we should probably ultimately do, but will be time consuming and (human) resource intensive. This is why I am suggesting we address it first with item 1, above.

If others feel that we should bite the bullet and spend the time to complete item 2, rather than implementing item 1, that's something we can do.

What do you think @willdurand, @diox and @muffinresearch?

**Comment by bobsilverberg** [comment]
FWIW, I was looking more closely at how Code Manager renders its Compare page, and it looks like it also uses the current version of the file (rather than the information coming back from `compare_to`) to render the File Info area as well. The way it deals with and stores data coming back from the `compare_to` endpoint is problematic and will need to be changed to support https://github.com/mozilla/addons-code-manager/issues/1599, so perhaps it does make sense to try to implement option 2 at this time.

**Comment by bobsilverberg** [comment]
@willdurand and I discussed this today, and he pointed out that if rendering of the code view area isn't blocked by this additional API request then maybe we don't need to worry about it at all. We're mostly concerned with performance of the code view area, and allowing a user to start viewing the code as soon as possible, so a delay in the rendering of the code overview isn't such a big deal. 

This, combined with the fact that changing the code overview to work with both file content and a diff is a significant amount of work, suggests that we should just ignore the issue for now.

I will verify that this additional API request does not block rendering of the code view area.

**Comment by bobsilverberg** [comment]
I believe this is no longer needed after the landing of https://github.com/mozilla/addons-code-manager/commit/d93411d062d6fe377ced792d3f228c434766c8c4
